playwright/flaky-reporter.ts

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2024 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Flaky test reporter, creating & updating GitHub issues
+ * Only intended to run from within GitHub Actions
+ */
+
+import type { Reporter, TestCase } from "@playwright/test/reporter";
+
+const REPO = "element-hq/element-web";
+const LABEL = "Z-Flaky-Test";
+const ISSUE_TITLE_PREFIX = "Flaky playwright test: ";
+
+class FlakyReporter implements Reporter {
+    private flakes = new Set<string>();
+
+    public onTestEnd(test: TestCase): void {
+        const title = `${test.location.file.split("playwright/e2e/")[1]}: ${test.title}`;
+        if (test.outcome() === "flaky") {
```

Comment:
could we also do this on actual failures?

Reply:
Actual failures get fed into the releases room to be dealt with immediately, this only concerns runs on the main branch as otherwise they could be introduced by the PR(s)

Reply:
ok, but isn't it the case that half the time "dealing with them" means re-running the test suite (and manually creating a tracking issue)?

Reply:
Yes, but if a test fails it isn't inherently flaky, and creating a solution which tracks manual retries is significantly harder, so leaving that to the Maintainer to handle

Reply:
I kinda thought that it would be easier for the issue to be opened automatically (and then closed manually, if necessary) than it would for the maintainer to have to open the issue manually.

But fair enough. I don't have strong feelings, and this will probably catch most useful cases.

Reply:
Perhaps, the intent is to catch the flakes that we miss in our weekly calls by just inspecting the report of the latest run